LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2Ability_LW_AssaultAbilitySet.uc

```diff
@@ -635,3 +639,154 @@ static function X2AbilityTemplate CreateChainLightningAbility()
+	local X2AbilityCooldown					Cooldown;	
+	local X2Condition_UnitType				ImmuneUnitCondition;
+	local X2Condition_UnitEffects			SuppressedCondition;
+	local X2Effect_ApplyWeaponDamage		DamageEffect;
+	//local X2Effect_RemoveEffects			CleanUpEffect;
+
+	`CREATE_X2ABILITY_TEMPLATE(Template, 'ChainLightning');
+	Template.ShotHUDPriority = class'UIUtilities_Tactical'.const.CLASS_COLONEL_PRIORITY;
+	Template.AbilitySourceName = 'eAbilitySource_Perk';
+	Template.eAbilityIconBehaviorHUD = eAbilityIconBehavior_AlwaysShow;
+	Template.IconImage = "img:///UILibrary_LW_Overhaul.LW_AbilityChainLightning";
+	Template.bCrossClassEligible = false;
+	Template.Hostility = eHostility_Offensive;
+	Template.AbilityTriggers.AddItem(default.PlayerInputTrigger);
+	Template.ActivationSpeech = 'TeslaCannon';
+
+	Template.AbilityShooterConditions.AddItem(default.LivingShooterProperty);
+	Template.AbilityTargetConditions.AddItem(default.LivingTargetUnitOnlyProperty);
+	Template.AbilityTargetConditions.AddItem(default.GameplayVisibilityCondition);
+	
+	Template.AbilityTargetStyle = default.SimpleSingleTarget;
+	Template.AbilityMultiTargetStyle = new class'X2AbilityMultiTarget_Volt';
+	Template.AbilityTriggers.AddItem(default.PlayerInputTrigger);
+	Template.TargetingMethod = class'X2TargetingMethod_Volt';
+
+	Template.AbilityShooterConditions.AddItem(default.LivingShooterProperty);
+	Template.AddShooterEffectExclusions();
+
+	Template.AbilityTargetConditions.AddItem(default.GameplayVisibilityCondition);
+	Template.AbilityTargetConditions.AddItem(default.LivingHostileUnitDisallowMindControlProperty);
+
+
+	//Template.bAllowAmmoEffects = true;
+	Template.bAllowBonusWeaponEffects = true;
+
+	// Can't target dead; Can't target friendlies -- must be enemy organic
+	UnitPropertyCondition = new class'X2Condition_UnitProperty';
+	UnitPropertyCondition.ExcludeRobotic = true;
+	UnitPropertyCondition.ExcludeOrganic = false;
+	UnitPropertyCondition.ExcludeDead = true;
+	UnitPropertyCondition.ExcludeFriendlyToSource = true;
+	UnitPropertyCondition.RequireWithinRange = true;
+	Template.AbilityTargetConditions.AddItem(UnitPropertyCondition);
+	
+	UnitPropertyCondition2=new class'X2Condition_UnitProperty';
+	UnitPropertyCondition2.ExcludeConcealed = true;
+	Template.AbilityShooterConditions.AddItem(UnitPropertyCondition2);
+
+	SuppressedCondition = new class'X2Condition_UnitEffects';
+	SuppressedCondition.AddExcludeEffect(class'X2Effect_Suppression'.default.EffectName, 'AA_UnitIsSuppressed');
+	SuppressedCondition.AddExcludeEffect(class'X2Effect_AreaSuppression'.default.EffectName, 'AA_UnitIsSuppressed');
+	Template.AbilityShooterConditions.AddItem(SuppressedCondition);
+
+	ImmuneUnitCondition = new class'X2Condition_UnitType';
+	ImmuneUnitCondition.ExcludeTypes.AddItem('PsiZombie');
+	ImmuneUnitCondition.ExcludeTypes.AddItem('AdvPsiWitchM2');
+	ImmuneUnitCondition.ExcludeTypes.AddItem('AdvPsiWitchM3');
+	Template.AbilityTargetConditions.AddItem(ImmuneUnitCondition);
```

Comment:
Should these be multi-target conditions as well?

Reply:
yeah, you're right.